### ISSUE TITLE
Add more information (success, failure, etc.)

### ISSUE THREAD

**Issue by hugandr** [PatWie/CppNumericalSolvers#33]
A few commits ago, there was already added an info struct, which allows to query some variables.

I think we should also add some more infos about the result of the minimization. Up until now, you don't even now whether the algorithm converged or not. 

So I suggest to add at least the information whether the algorithm was successfull. If it wasn't, some information should also be provided. E.g. NO_CONVERGENCE, LEFT_DOMAIN, etc.

I also experienced a case, where the solver had a smaller minimum at one point and later got worse. Maybe it would also be better to store the overall minimum + location and return this one as the minimum and not the last point. 
This would also make sense in some applications, where you need to get back a valid position/value in any cases.

These are just a few small suggestions that came to my mind when using the library. Feel free to correct me if you don't agree!

**Comment by PatWie** [comment]
Everything fine. That is usefull stuff. At the weekend I will have some time to look at it. There is also the open "gtest"-branch.

Thanks for all these suggestions!

**Comment by spinicist** [comment]
I made a start on this in #34. Hope it is helpful.

**Comment by PatWie** [comment]
This feature was implemented by @spinicist  in #34. Cool!

**Comment by PatWie** [comment]
However, this introduced some issues. See the 'gtest' branch and the example "simple.cpp".

**Comment by spinicist** [comment]
I just pulled your gtest branch and simple.cpp compiles fine, but cmaes and neldermead don't.

I think the merges between different branches have ended up with some stuff in the wrong order. I think the root cause is I rebased #34 off the gtest branch, which you made before #29 so those commits are missing.

I will have a look later today at rebasing stuff to get it all in the correct order, and make sure there's an example for the Criteria in simple.cpp.
